crates/ruff_linter/src/rules/pygrep_hooks/rules/blanket_noqa.rs

```diff
@@ -85,7 +86,19 @@ pub(crate) fn blanket_noqa(
     diagnostics: &mut Vec<Diagnostic>,
     noqa_directives: &NoqaDirectives,
     locator: &Locator,
+    exemption: &Option<FileExemption>,
```

Comment:
One downside here is that this is only going to flag the _first_ `# ruff: noqa` in the file. I assume if you add multiple such comments, only the first is flagged as a diagnostic?

Reply:
We might need to do something like `NoqaDirectives`, whereby we parse all of these upfront, then use the parsed exemptions in `crates/ruff_linter/src/checkers/noqa.rs` (and here).

Reply:
@ajesipow - Do you want to give that a try?

Reply:
Thanks for the ping @charliermarsh.
I haven't found the time yet after work this week, but today is a public holiday over here so I'll look into it today :)

Reply:
I just pushed the changes and adapted the tests again accordingly.

Looking forward to your feedback again.